keras_nlp/models/bert/bert_backbone.py

```diff
@@ -190,28 +190,28 @@ def __init__(
             },
             **kwargs,
         )
+
         # All references to `self` below this line
         self.vocabulary_size = vocabulary_size
-        self.hidden_dim = hidden_dim
-        self.intermediate_dim = intermediate_dim
         self.num_layers = num_layers
         self.num_heads = num_heads
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
         self.max_sequence_length = max_sequence_length
         self.num_segments = num_segments
-        self.dropout = dropout
-        self.token_embedding = token_embedding_layer
```

Comment:
We need this for our bert pretraining example, when using the `token_embedding` weight matrix to reverse from hidden_dim -> vocabulary logits.

If we have another way (e.g. a layer lookup by name on the class that is stable across multiple model instantiations) that could be fine, but we should make the accompanying change in `examples/bert_pretraining` too. Note that this is just example code right now, but we will be adding MaskedLM tasks that need this inside the library itself.

Reply:
+1. We need a hook into the token embedding, so it was this or a `gimme_that_token_embedding()` method.

Reply:
Can we do an encoder.get_layer(...) here? Will that work?

Reply:
The only issue (in the future) with that is that the name of the embedding layer will have to be `"token_embedding"`. However, across all our model offerings, the name is `"token_embedding"`...so, probably not an issue. 

I've changed it to `model.get_layer("token_embedding").embeddings`. Seems to work fine: https://colab.research.google.com/drive/1-vxwtEC1Ipne3oVQBDbSBYncGW9ILqK3?usp=sharing.

Reply:
Yeah, that will work fine I think. The question is more if we want our layer naming to be API stable. I worry it will sneak up on people--no one will associate the layer name with an API contract.

I might go actually go heavy handed here, and add a `token_embedding` documented property, potentially at the backbone level, and actually document this on keras.io. Doing "stuff" with the token embedding seems fairly common, so giving a hook to it at the API level might make sense, and generally be a friendly thing to do.

This would also give a uniform way to access across backbones. E.g. Roberta uses our `TokenAndPositionEmbedding`, so for that backbone you would need `backbone.get_layer("embeddings").token_embedding.embeddings` ooph.

I'm not opposed to this though, as we could always grow into the heavy handed solution. cc @jbischof for thoughts.

Reply:
Are you proposing exposing it as another `@property`? Fine by me

Reply:
yeah, in `Backbone`

```
@property
def token_embedding(self):
    """A `keras.layers.Embedding` instance for embedding token ids."""
    return self._token_embedding
```

With the plan to document and export this. Given how common it is to do stuff with the token embedding in particular, this would make sense to me to be in the backbone contract.

Reply:
lgtm

Reply:
Gotcha, making changes!